prototypes/generator.lua

```diff
@@ -0,0 +1,216 @@
+  },
+  { -- Not in starting area
+    influence = -5,
+    starting_area_weight_optimal = 1,
+    starting_area_weight_range = 0,
+    starting_area_weight_max_range = 0
+  },
+  {
+    influence = 100, -- ... except for starting tile
+    min_influence = 0,
+    distance_optimal = 0,
+    distance_range = 0.1,
+    distance_max_range = 0.1
+  },
+  {
+    influence = -5
+  }
+}}
+
+-- highground sand
+data.raw.tile['sand-2'].autoplace = {
+  peaks = {
+  {
+    influence = 5,
+    min_influence = 0,
+    elevation_optimal = scale_elevation(15),
+    elevation_range = 5 * elevation_scale,
+    elevation_max_range = 5 * elevation_scale,
+    tier_from_start_optimal = 0.1,
+    tier_from_start_range = 0,
+    tier_from_start_max_range = 0,
+    tier_from_start_top_property_limit = 0.1
+  },
+  {
+    influence = 1,
+    max_influence = 0,
+    starting_area_weight_optimal = 1,
+    starting_area_weight_range = 0,
+    starting_area_weight_max_range = 0
+  },
+  {
+    influence = -100, -- not on starting tile
+    max_influence = 0,
+    distance_optimal = 0,
+    distance_range = 3,
+    distance_max_range = 3
+  },
+}}
+
+data.raw.tile['water'].autoplace = {
+  peaks = {
+  {
+    influence = 0.1, -- shallow water around cliff islands
+    min_influence = 0,
+    elevation_optimal = -2 * elevation_scale + waterline,
+    elevation_range = 2.5 * elevation_scale,
+    elevation_max_range = 2.5 * elevation_scale
+  },
+  {
+    influence = 5, -- around starting tile
+    min_influence = 0,
+    distance_optimal = 0,
+    distance_range = 5,
+    distance_max_range = 5
+  }
+}}
+
+data.raw.tile['deepwater'].autoplace = {
+  peaks = {
+  {
+    influence = 0.01
+  }
+}}
+
+local noise = require "noise"
+local tne = noise.to_noise_expression
+local enemy_random_seed = 1
+local function new_random_seed()
+  enemy_random_seed = enemy_random_seed + 1
+  return enemy_random_seed
+end
+local function worm_autoplace(distance, probability, order, falloff)
+  local d = noise.var("distance") - noise.var("starting_area_radius")
+  p = noise.clamp((d - distance * 128) / 128, 0, 1)
+  if falloff then
+    p = p * noise.clamp(((distance + 2) * 128 - d) / 128, 0, 1)
+  end
+  p = p * noise.clamp((waterline - noise.var("elevation")), 0, 1)
+  p = p * probability
+  p = noise.random_penalty(p, probability * 0.5, {
+    x = noise.var("x") + new_random_seed()
+  })
+
+  return {
+    order = order,
+    force = "enemy",
+    probability_expression = p,
+    richness_expression = tne(1)
+  }
+end
+
+data:extend({
+  {
+    type = 'noise-layer',
+    name = 'desert-tree-noise'
+  },
+  {
+    type = 'noise-layer',
+    name = 'temperate-tree-noise'
+  },
+  {
+    type = 'noise-layer',
+    name = 'swamp-tree-noise'
+  },
+  {
+    type = 'noise-layer',
+    name = 'desert-garden-noise'
+  },
+  {
+    type = 'noise-layer',
+    name = 'temperate-garden-noise'
+  },
+  {
+    type = 'noise-layer',
+    name = 'swamp-garden-noise'
+  }
+})
+--[[
```

Comment:
What is that?

Reply:
Good catch, a portion of this was dead code that need to be removed.